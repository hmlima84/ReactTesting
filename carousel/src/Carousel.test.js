import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Carousel from './Carousel'; // Import your Carousel component
import TEST_IMAGES from "./_testCommon.js"; // Import test images or create a sample array

/* Define test images
const TEST_IMAGES = [
  { src: "test1.com", caption: "testing image 1" },
  { src: "test2.com", caption: "testing image 2" }
];
*/

//smoke test
it('renders without crashing', () => {
  const photos = [{ src: 'image-src-1', caption: 'Image 1' }, { src: 'image-src-2', caption: 'Image 2' }];
  render(<Carousel photos={photos} title="Test Title" />);
});

//snapshot test
it('matched card', () => {
  const photos = [{ src: 'image-src-1', caption: 'Image 1' }, { src: 'image-src-2', caption: 'Image 2' }];
  const {asFragment} = render(<Carousel photos={photos} title="Test Title" />);
  expect(asFragment()).toMatchSnapshot()
})

// Test case
it("works when you click on the right arrow", () => {
  const { container } = render(
    <Carousel photos={TEST_IMAGES} title="images for testing" />
  );

  // expect the first image to show, but not the second
  expect(
    container.querySelector('img[alt="testing image 1"]')
  ).toBeInTheDocument();
  expect(
    container.querySelector('img[alt="testing image 2"]')
  ).not.toBeInTheDocument();

  // move forward in the carousel
  const rightArrow = container.querySelector(".bi-arrow-right-circle");
  fireEvent.click(rightArrow);

  // move backward in the carousel
  const leftArrow = container.querySelector(".bi-arrow-left-circle");
  fireEvent.click(leftArrow);

  // expect the second image to show, but not the first
  expect(
    container.querySelector('img[alt="testing image 1"]')
  ).not.toBeInTheDocument();
  expect(
    container.querySelector('img[alt="testing image 2"]')
  ).toBeInTheDocument(); 
}); 
