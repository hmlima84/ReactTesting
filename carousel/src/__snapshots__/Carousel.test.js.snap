// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matched card 1`] = `
<DocumentFragment>
  <div
    class="Carousel"
  >
    <h1>
      Test Title
    </h1>
    <div
      class="Carousel-main"
    >
      <i
        class="bi bi-arrow-left-circle"
      />
      <div
        class="Card"
      >
        <h4
          class="Card-title"
        >
          Image 1
        </h4>
        <img
          alt="Image 1"
          class="Card-image"
          src="image-src-1"
        />
        <small
          class="Card-small"
        >
          Image 1 of 2.
        </small>
      </div>
      <i
        class="bi bi-arrow-right-circle"
      />
    </div>
  </div>
</DocumentFragment>
`;
